tests/test_issue_43_dishwasher_passive_drying.py

```
@@ -369,3 +411,160 @@ def test_completed_cycle_end_time_not_set_to_terminal_spike(self) -> None:
+
+class TestEndSpikeProgressGate:
+    """Regression tests for the issue #43 end-spike progress gate.
+
+    Reproduces the full April 27 user scenario from
+    home-assistant_ha_washdata_2026-04-27T21-08-33.248Z.log:
+
+      18:08:35  STARTING → RUNNING (active wash, ~85W, then 2000W heating)
+      19:57:34  power drops to 0W → RUNNING → PAUSED
+      19:57:44  PAUSED → ENDING                  (~109 min, 47% of expected)
+      20:06:35  high-power spikes in ENDING       (~117 min, 50% of expected)
+      ...       passive drying at 0–0.5W          (~117 min → 234 min)
```

Ruff (0.15.12)

[warning] Docstring contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF002)